### PR TITLE
hidapi: bump version

### PR DESCRIPTION
**Pull request by AlwaysLivid**
Relevant issue: #4496

The recipe cannot be used as of 7fc94ca.

**TO-DO:**
* Remove old recipe (?)
* ~Check if ./bootstrap is actually necessary (?)~
* ~Update/Prepare patches for the latest version~

### CODE REVIEW

File: dev-libs/hidapi/hidapi-0.9.0.recipe [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+HOMEPAGE="http://www.signal11.us/oss/hidapi/"
+COPYRIGHT="2010 Alan Ott
+	2011 Signal 11 Software"
+LICENSE="BSD (3-clause)
+	GNU GPL v3"
+REVISION="2"
+SOURCE_URI="https://github.com/libusb/hidapi/archive/hidapi-$portVersion.tar.gz"
+CHECKSUM_SHA256="630ee1834bdd5c5761ab079fd04f463a89585df8fcae51a7bfe4229b1e02a652"
+SOURCE_DIR="hidapi-hidapi-$portVersion"
+# PATCHES="hidapi-$portVersion.patchset"
```
Comment (extrowerk) [inline dev-libs/hidapi/hidapi-0.9.0.recipe]: Remove if unneeded

Comment (alwayslivid) [reply dev-libs/hidapi/hidapi-0.9.0.recipe]: Not yet, it is needed and needs to be worked on.

**Comment by alwayslivid** [comment]
Compiles successfully on x86_64. I checked out the patch for any sort of redundant or outdated code-- looks good to me.

**Comment by extrowerk** [comment]
Build ok on 64 bit but test fails win missing symbol libicon_close

**Comment by extrowerk** [comment]
Whats wit the todo-s?

**Comment by alwayslivid** [comment]
> Whats wit the todo-s?

I'm not quite sure as to whether I should remove the older recipe from 2016. I'm leaning towards "I should", but I'd like to get a confirmation.



> Build ok on 64 bit but test fails win missing symbol libicon_close

Mind sending a build log?

Here's a [random, possibly relevant StackOverflow thread](https://stackoverflow.com/questions/52364422/c-convert-iso-iso%E2%88%928859-1-to-utf-8-with-libconv).

**Comment by extrowerk** [comment]
You should test the dependant recipes, and if they builds/works, then you can delete the old one.
You are using libiconv in this recipe, so yep.

**Comment by extrowerk** [comment]
http://0x0.st/irhG.txt

**Comment by alwayslivid** [comment]
> You should test the dependant recipes, and if they builds/works, then you can delete the old one.
> You are using libiconv in this recipe, so yep.

Oh wait, libiconv has already been ported and is used by the recipe, missed that one.

There seems to be a previous mention of that error by @Begasus in the [IRC logs](https://echelog.com/logs/browse/haiku/1577487600) alongside some silly, completely irrelevant questions that I made.

According to the build log, the 2016 version has been built rather than hidapi-0.9.0, however. @extrowerk

**Comment by extrowerk** [comment]
2016 >> 0.9.0, so you definetely should delete the old one. thanks for pointing out, doing again...

**Comment by extrowerk** [comment]
Same error during test with 0.9.0

**Comment by alwayslivid** [comment]
> Same error during test with 0.9.0

Shouldn't be too big of a deal from a functional perspective if lots of recipes depend on lib:libhidapi that shows the same error, although it'd be nice to get it shorted out. Not sure what's the root cause of it, but I'll try to look into it.

**Comment by alwayslivid** [comment]
> ...although it'd be nice to get it shorted out.

Would it be a good idea to open up an issue for that?

**Comment by Begasus** [comment]
> http://0x0.st/irhG.txt

Got the same error here @extrowerk 
PS, did some work on a fork I created here: https://github.com/Begasus/hidapi/commit/1435807a6eb84a4775a4aefb7a41501fb6d9dab3

**Comment by Begasus** [comment]
```
make hidtest
```
Is actually already done in BUILD
Try to add in BUILD,
```
export LDFLAGS="-liconv"
```

**Comment by alwayslivid** [comment]
> > http://0x0.st/irhG.txt
> 
> Got the same error here @extrowerk
> PS, did some work on a fork I created here: [Begasus/hidapi@1435807](https://github.com/Begasus/hidapi/commit/1435807a6eb84a4775a4aefb7a41501fb6d9dab3)

Two questions:
* Should your fork be used over the upstream version?
* Would it be a good idea to propose the changes to the upstream version?

**Comment by Begasus** [comment]
> > PS, did some work on a fork I created here: [Begasus/hidapi@1435807](https://github.com/Begasus/hidapi/commit/1435807a6eb84a4775a4aefb7a41501fb6d9dab3)
> 
> Two questions:
> 
>     * Should your fork be used over the upstream version?
> 
>     * Would it be a good idea to propose the changes to the upstream version?

That was my first intention :) But haven't looked into recent changes

**Comment by alwayslivid** [comment]
For the sake of sustainability, I proposed @CodeforEvolution's patches to be merged in the upstream tree of hidapi.

https://github.com/libusb/hidapi/pull/141

**Comment by Begasus** [comment]
Don't think the changes to the patchset are relavent, also, a few lines seem off ("if OS_KFREEBSD" (needed? already done in the prior patch/section), change to CPPOBJS to CPPOBJ and later still use CPPOBJS?, move for "$" in the "shell finddir"?)
All in all, if the patchset applies clean it should be kept as is I think ...

**Comment by Begasus** [comment]
@AlwaysLivid 
https://github.com/libusb/hidapi/compare/master...Begasus:haiku#diff-67e997bcfdac55191033d57a16d1408aR123
This should fix not having to use "export LDFLAGS = -liconv"

**Comment by AlwaysLivid** [comment]
@Begasus, there were a few WIP changes that also involved Haiku in the Makefile.

The changes weren't made by me, but since you're bringing it up, should I include that change in this PR?

**Comment by Begasus** [comment]
@AlwaysLivid haven't checked but maybe you could just rename the patchset?

**Comment by Begasus** [comment]
https://github.com/Begasus/hidapi/commit/74b3719f01e9a8efd0bfdd4a16e405b3c3bd3e2a

**Comment by Begasus** [comment]
PS, already commented on the open PR at hidapi: https://github.com/libusb/hidapi/pull/30

**Comment by Begasus** [comment]
Among some of the issues I found relating this: https://trac.osgeo.org/mapserver/ticket/909#comment:3

**Comment by AlwaysLivid** [comment]
Now that the [changes](https://github.com/libusb/hidapi/pull/141) have been merged to the upstream source, the patches are no longer necessary.

This repository will be updated accordingly, but I think it would be a good idea if we could wait until the 0.10 release, which isn't too far away, considering that there are only [a few bugs](https://github.com/libusb/hidapi/projects/1#card-32840833) that need to be fixed. However, one of those bugs seems pretty nasty themselves for them to be implemented in upstream Haiku, so I was wondering whether waiting on it would be a good idea.

**Comment by Begasus** [comment]
Maybe wait until 0.10 is released, looking at the open issues they don't seem to be Haiku related?

**Comment by AlwaysLivid** [comment]
> Maybe wait until 0.10 is released, looking at the open issues they don't seem to be Haiku related?

They aren't, I'll work on this later. 

Just assumed that it'd be a good idea to just seamlessly transition from fetching repositories to the ordinarily versioned instances of the program.

**Comment by AlwaysLivid** [comment]
WIll open a new pull request for this.

**Pull request closed without merging**